src/core/Entry.cpp

```diff
@@ -355,11 +367,20 @@ QString Entry::totpSeed() const
     m_data.totpStep = QTotp::defaultStep;
 
     if (m_attributes->hasKey("TOTP Settings")) {
-        QRegExp rx("(\\d+);(\\d)", Qt::CaseInsensitive, QRegExp::RegExp);
+        QString digitsRx = "(?:\\d+)";
+        // build a regex that supports any encoders that we support.
+        foreach (QTotp::Encoder encoder, QTotp::encoders) {
+            digitsRx.append("|").append(encoder.shortName);
+        }
+        QRegExp rx(QString("(\\d+);(%1)").arg(digitsRx));
```

Comment:
Use QRegularExpression, not QRegExp. I'm also not too big a fan of dynamically assembled regular expressions. The can easily introduce bugs and even security issues if not escaped properly.